src/windows/audio.c

```diff
@@ -331,16 +334,28 @@ void MTY_AudioDestroy(MTY_Audio **audio)
 	*audio = NULL;
 }
 
-static uint32_t audio_get_queued_frames(MTY_Audio *ctx)
+static HRESULT audio_get_queued_frames(MTY_Audio *ctx, uint32_t *out)
 {
-	if (!ctx->client)
-		return ctx->buffer_size;
+	HRESULT e = S_OK;
+	uint32_t frames = ctx->buffer_size; // indicates that the buffer is full, which also serves
+	                                    // as a useful return value in the case of an error.
+
+	if (!ctx->client) {
+		e = AUDCLNT_E_NOT_INITIALIZED;
+		goto except;
+	}
 
 	UINT32 padding = 0;
-	if (IAudioClient_GetCurrentPadding(ctx->client, &padding) == S_OK)
-		return padding;
+	e = IAudioClient_GetCurrentPadding(ctx->client, &padding);
+	if (e != S_OK)
+		goto except;
 
-	return ctx->buffer_size;
+	frames = padding;
+
+	except:
+
+	*out = frames;
+	return e;
```

Comment:
```suggestion

	return e;
```

frames isn't needed, since it will be set in all scenarios.

Reply:
Fair enough, the goto-except thing is probably overkill. I'll rewrite this taking your comments into account, let me know what you think when you see it.